### ISSUE TITLE
Problem: Unable to Execute Multiple Tasks in Parallel When Sharing a Template

### ISSUE THREAD

**Issue by nodemasterpro** [ansible-semaphore/semaphore#2561]
### Issue

I am experiencing an issue with Semaphore where tasks that share the same template cannot run in parallel, even when they target different hosts. Despite using a static local repository and enabling concurrency_mode and max_parallel_tasks, Semaphore continues to queue tasks instead of running them simultaneously.

### Impact

Ansible (task execution)

### Installation method

Docker

### Database

Postgres

### Semaphore Version

Version: 2.10.43

### Ansible Version

```bash
Ansible version: 2.16.13
```

### Logs & errors

 3:48:46 PM
Task 1030 added to queue

### Manual installation - system information

N/A

### Configuration

Concurrency Mode: node
Max Parallel Tasks: Configured in the WebUI.

### Additional information

I have confirmed that switching to static local repositories resolved previous Git synchronization issues but did not enable parallel execution for tasks sharing a template. This limitation might be related to template locking logic.

Request:
Please consider offering a solution or guidance on enabling tasks to run in parallel when they share a template. Options such as disabling template locking for static repositories or a more granular locking mechanism would significantly improve usability in high-concurrency environments.

**Comment by Guits32** [comment]
@nodemasterpro Did you find any solution or support for this?

**Comment by nodemasterpro** [comment]
@Guits32 Yes, I managed to work around the problem by removing the limitations in the source code (TaskPool.go file). I recompiled. It works.
